src/nvim/msgpack_rpc/channel.c

```diff
@@ -823,6 +823,7 @@ static void log_server_msg(uint64_t channel_id,
   msgpack_unpack_next(&unpacked, packed->data, packed->size, NULL);
   uint64_t type = unpacked.data.via.array.ptr[0].via.u64;
   DLOGN("[msgpack-rpc] nvim -> client(%" PRIu64 ") ", channel_id);
+  log_lock();
```

Comment:
Possibly if this is moved to log.c, but I don't think it's worth the trouble because the usage of log.c mutex is really simple and won't spread beyond this and the standard log function.